### PR TITLE
add kpm pkg include and exclude

### PR DESCRIPTION
**Pull request by officialasishkumar**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y #306  

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

**Comment by officialasishkumar** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Peefy** [comment]
> Hey @Peefy,
> 
> In [kpm/pkg/package/toml_test.go](https://github.com/kcl-lang/kpm/blob/b8a87ca8d28cccf56777ec1ad3725b6d9de3b236/pkg/package/toml_test.go#L74), I'm not able to unmarshal the array of strings. Can you please help me on why it's not working? I tried converting the include array to a string and it's working fine for it.

cc @zong-zhe

**Comment by officialasishkumar** [comment]
@Peefy PTAL

File: pkg/package/toml_test.go [hunk 1]
```diff
@@ -77,6 +77,8 @@ func TestUnMarshalTOML(t *testing.T) {
 	assert.Equal(t, modfile.Pkg.Name, "MyKcl")
 	assert.Equal(t, modfile.Pkg.Edition, "v0.0.1")
 	assert.Equal(t, modfile.Pkg.Version, "v0.0.1")
+	assert.Equal(t, modfile.Pkg.Include, []string{"src/", "README.md", "LICENSE"})
+    assert.Equal(t, modfile.Pkg.Exclude, []string{"target/", ".git/", "*.log"})
```
Comment (Peefy) [inline pkg/package/toml_test.go:81]: Format the document.

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -126,7 +126,7 @@ func Exists(path string) (bool, error) {
 // todo: Consider using the OCI tarball as the standard tar format.
 var ignores = []string{".git", ".tar"}
 
-func TarDir(srcDir string, tarPath string) error {
+func TarDir(srcDir string, include []string, exclude []string, tarPath string) error {
```
Comment (Peefy) [inline pkg/utils/utils.go:129]:
```suggestion
func TarDir(srcDir string, tarPath string, include []string, exclude []string) error {
```

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -137,6 +137,8 @@ func TarDir(srcDir string, tarPath string) error {
 	tw := tar.NewWriter(fw)
 	defer tw.Close()
 
+	fmt.Println(exclude)
```
Comment (Peefy) [inline pkg/utils/utils.go:140]: Remove this code.

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -148,6 +150,18 @@ func TarDir(srcDir string, tarPath string) error {
 			}
 		}
 
+		for _, ex := range exclude {
+			if strings.Contains(path, ex) {
```
Comment (Peefy) [inline pkg/utils/utils.go:154]: Here, file format matching should be used, such as `*.log`, instead of using string containing functions.

Comment (officialasishkumar) [reply pkg/utils/utils.go]: updated

File: pkg/package/toml_test.go [hunk 5]
```diff
@@ -77,6 +77,8 @@ func TestUnMarshalTOML(t *testing.T) {
 	assert.Equal(t, modfile.Pkg.Name, "MyKcl")
 	assert.Equal(t, modfile.Pkg.Edition, "v0.0.1")
 	assert.Equal(t, modfile.Pkg.Version, "v0.0.1")
+	assert.Equal(t, modfile.Pkg.Include, []string{"src/", "README.md", "LICENSE"})
```
Comment (Peefy) [inline pkg/package/toml_test.go:80]: Test failed.

File: pkg/utils/utils_test.go [hunk 6]
```diff
@@ -74,8 +74,8 @@ func TestTarDir(t *testing.T) {
 	if !os.IsNotExist(err) {
 		os.Remove(tarPath)
 	}
-
-	err = TarDir(filepath.Join(testDir, "test_src"), tarPath)
+	emptyArrayOfStrings := []string{}
+	err = TarDir(filepath.Join(testDir, "test_src"), tarPath, emptyArrayOfStrings, emptyArrayOfStrings)
```
Comment (Peefy) [inline pkg/utils/utils_test.go:78]: Could you please add more test cases when the include path and exclude paths are not empty.

Comment (officialasishkumar) [reply pkg/utils/utils_test.go]: updated, please check

File: pkg/package/toml_test.go (hunk 5 above)
Comment (officialasishkumar) [reply pkg/package/toml_test.go:80]: the test is passing on my machine 
image

Comment (Peefy) [reply pkg/package/toml_test.go:80]: OK. I will merge this PR first, and investigate it now.

**Review by Peefy — APPROVED** [review]
LGTM
